src/lib.rs

```diff
@@ -1,253 +1,253 @@
-    self.quiet = quiet;
-  }
-
-  fn thread_pool(&mut self) -> &mut futures::executor::ThreadPool {
-    if self.thread_pool.is_some() {
-      self.thread_pool.as_mut().unwrap()
-    } else {
-      let pool = match self.thread_count {
-        Some(count) => futures::executor::ThreadPool::builder().pool_size(count).create(),
-        None => futures::executor::ThreadPool::new(),
-      };
-
-      self.thread_pool = Some(pool.unwrap_or_else(|err| panic!("failed to create job pool: {}", err)));
-      self.thread_pool.as_mut().unwrap()
+
+    pub fn quiet(&mut self, quiet: bool) {
+        self.quiet = quiet;
     }
-  }
-
-  pub fn execute<T: Send + 'static, E: Send + 'static>(&mut self, mut job: Job<T, E>) -> ExecutionResults<T, E> {
-    let task_count = job.tasks.len();
-    let pb = Arc::new(
-      if self.quiet {
-        indicatif::ProgressBar::hidden()
-      } else {
-        indicatif::ProgressBar::new(task_count as u64)
-      }
-    );
-    pb.set_style(progress_bar_style(task_count));
-    pb.set_prefix(job.name.clone());
-    pb.enable_steady_tick(1000);
-
-    trace_scoped!(format!("Pool::execute({})", job.name));
-
-    let task_monitor = Arc::new(Mutex::new(TaskMonitor::new(Arc::clone(&pb))));
-    let handles: Vec<_> = job
-      .tasks
-      .drain(..)
-      .map(|task| {
-        let task_monitor = Arc::clone(&task_monitor);
-        self
-          .thread_pool()
-          .spawn_with_handle(future::lazy(move |context| {
-            trace_scoped!(task.name.clone());
-            let task_id = task_monitor.lock().unwrap().started(task.name.clone());
-            let result = (task.task)(&context);
-            task_monitor.lock().unwrap().finished(task_id);
-            (task.name, result)
-          }))
-          .expect("failed to spawn job")
-      })
-      .collect();
-
-    let handles = futures::executor::block_on(future::join_all(handles));
-    pb.finish();
-
-    let mut successful = Vec::new();
-    let mut failed = Vec::new();
-
-    for (task_name, result) in handles {
-      match result {
-        Ok(result) => {
-          successful.push(ExecutionResult {
-            name: task_name,
-            result,
-          });
-        }
 
-        Err(err) => failed.push(ExecutionResult {
-          name: task_name,
-          result: err,
-        }),
-      }
+    fn thread_pool(&mut self) -> &mut rayon::ThreadPool {
+        if self.thread_pool.is_some() {
+            self.thread_pool.as_mut().unwrap()
+        } else {
+            let builder = rayon::ThreadPoolBuilder::new();
+            let builder = match self.thread_count {
+                Some(count) => builder.num_threads(count),
+                None => builder,
+            };
+
+            self.thread_pool = Some(
+                builder
+                    .build()
+                    .unwrap_or_else(|err| panic!("failed to create job pool: {}", err)),
+            );
+            self.thread_pool.as_mut().unwrap()
+        }
```

Comment:
Better todo a `match` instead of `is_some()` and then `unwrap()`

```suggestion
       match self.thread_pool {
           Some(pool) => pool,
           None => {
               let builder = rayon::ThreadPoolBuilder::new();
               let builder = match self.thread_count {
                   Some(count) => builder.num_threads(count),
                   None => builder,
               };
               
           self.thread_pool = Some(
                builder
                    .build()
                    .unwrap_or_else(|err| panic!("failed to create job pool: {}", err)),
            );
            self.thread_pool.as_mut().unwrap()
           }
       }
```

Reply:
Did one better.